app/controllers/job_postings_controller.rb

```diff
@@ -1,47 +1,55 @@
 class JobPostingsController < ApplicationController
-
+  before_action :find_job_posting, only: [:show, :edit, :update, :destroy]
+  
   def index
     @job_postings = if current_employer
                       current_employer.job_postings
                     elsif current_employee
-                      current_employee.job_postings
+                      JobPosting.all
                     end
   end
 
   def new
     @job_posting = JobPosting.new
-    @job_posting.skills.build
   end
 
   def create
-    @job_posting = JobPosting.new(job_posting_params)
+    @job_posting = JobPosting.new(job_postings_params)
+    
     if @job_posting.save
-      redirect_to root_path
+      redirect_to @job_posting
     else
-      @job_posting.skills.build if @job_posting.skills.blank?
-      render :new
+      render "New"
```

Comment:
Better convention to use a symbol rather than a string:

`render :new` instead of `render "New"`